src/erlcloud_aws.erl

```diff
@@ -931,7 +931,9 @@ get_availability_zone() ->
 cache_instance_metadata_availability_zone() ->
     % it fine to use default here - no IAM is used, only for http client
     % one cannot use auto_config()/default_cfg() as it creates an infinite recursion.
-    case erlcloud_ec2_meta:get_instance_metadata("placement/availability-zone", #aws_config{}) of
+    Config = #aws_config{},
+    IMDSv2Token = maybe_imdsv2_session_token(Config),
```

Comment:
are you saying this should ALWAYS be the new default behaviour?

Reply:
Yes.. IMDSv2 should be the default behaviour due to security concerns. you can check https://www.uptycs.com/blog/you-should-be-using-aws-imdsv2-heres-why-and-how-to-do-it

Reply:
Not requiring IMDSv2 is currently consider a high severity issue. In order to required IMDSv2 in EC2, all the software should be compliant.
Currently, from our test, we know that at least erlcloud_ddb2 will fail when trying to access dynamodb.

Reply:
This is the official AWS documentation about the path to transition to IMDSv2

Reply:
cool. let me read little more about it too. 
but why not ask just right here - do IAM roles attached to host need new permissions?